text/0000-native-cli.md

```diff
@@ -0,0 +1,136 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some users report liking the `neon` CLI. But we don't actually need to eliminate it, and should at least continue to support it for some time in order to avoid disrupting users. Over time, we can decide whether it's becoming less popular. But there's really no need to couple the decision of what to do with `neon` to the decision of whether to do this RFC. Deprecation and removal is a separate question.
+
+There's some loss of simplicity in the source code from `require('addon')` to `process.dlopen(/* ..find the DLL.. */)`. We could perhaps abstract this code into a Neon library, but it may not even be necessary. It's still relatively straightforward and not really a part of the code that users touch anyway.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+## Alternative: Only use `cargo`
+
+We could build all the tooling through Cargo generators (e.g. with the [cargo-generate](https://github.com/ashleygwilliams/cargo-generate) Cargo addon). This would double down on the idea that Neon is about using Rust in your Node projects.
+
+However, on the outside, a Neon project _is_ a JS package, and you still need to use npm e.g. to install dependencies or to publish to the registry. So it makes sense to use `npm init` for creating a Neon project and `npm i` for building the project (including fetching and building any dependencies it may have). However, it's still useful for `cargo build` to work in the root directory for convenience, especially for cases where the user may add additional build logic to `npm install` but still want to be able to just build the Rust code by itself.
+
+We could use an `@neon` namespace instead of `create-neon-lib` etc. However, this makes for a noisier and less convenient-to-type command-line experience:
+```
+npm init @neon/app my-app
+```
+(Note that as a _programming_ syntax, this might arguably be preferable for the visual distinction. But the command-line is meant to be quick and easy to type and remember.) Another option would be to support both as aliases for each other, but this might just overwhelm users with pointless choices.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- What other build invalidation logic, if any, do we still need to add? Can this all be done within `build.rs`? We should review the `neon build` invalidation logic and check for anything this RFC has overlooked.
+- What is the best way for `cargo build` to work in the root?
+  * `cargo build --all`?
+  * `cargo build` by shelling out to `cargo build` in the right subdirectory?
+  * Flatten the directory structure so the root directory is both a Node package and a Rust crate? (This is an attractive option, as it simplifies the directory structure e.g. in the tree view of an IDE, and generally may make the cognitive load of a Neon project feel lighter.)
```

Comment:
👍  I like this the best. More complex project layouts can use a mono-repo to be more expressive. For example, instead of TypeScript and Rust needing to content for the coveted `src` directory, they end up moved into separate modules.

Reply:
Latest draft proposes and describes the flattened structure.